mcs/class/System.Private.CoreLib/System/GC.cs

```diff
@@ -2,8 +2,244 @@
 
 namespace System
 {
+	[Serializable]
+	public enum GCCollectionMode
+	{
+		Default = 0,
+		Forced = 1,
+		Optimized = 2
+	}
+
+	[Serializable]
+	public enum GCNotificationStatus
+	{
+		Succeeded = 0,
+		Failed = 1,
+		Canceled = 2,
+		Timeout = 3,
+		NotApplicable = 4
+	}
+
 	partial class GC
 	{
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		extern static int GetCollectionCount (int generation);
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		extern static int GetMaxGeneration ();
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		extern static void InternalCollect (int generation);
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		extern static void RecordPressure (long bytesAllocated);
+
+		// TODO: Move following to ConditionalWeakTable
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		internal extern static void register_ephemeron_array (Ephemeron[] array);
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		extern static object get_ephemeron_tombstone ();
+
+		internal static readonly object EPHEMERON_TOMBSTONE = get_ephemeron_tombstone ();
+
+		internal static void GetMemoryInfo(out uint highMemLoadThreshold,
+												  out ulong totalPhysicalMem,
+												  out uint lastRecordedMemLoad,
+												  // The next two are size_t
+												  out UIntPtr lastRecordedHeapSize,
+												  out UIntPtr lastRecordedFragmentation)
+		{
```

Comment:
This API was reworked, made public and moved to shared partition:
https://github.com/dotnet/corert/commit/0227127ff610d2f4c0b8b87b38ecb40366311560

Reply:
I should have been clearer. There's still one non-shared method that is needed:
`public static GCMemoryInfo GetGCMemoryInfo() => default;`

Reply:
We have that.

Reply:
Ah, sorry. I missed it. It was just behind the edge of the GitHub diff :)